hs_app_timeseries/templates/pages/timeseriesresource.html

```diff
@@ -217,6 +244,303 @@ <h3 style="color: red" id="file-type-error"></h3>
+                            $('#id-site').find("button.btn-primary").show();
+                        }
+                        else {
+                            $('#id-site').find("button.btn-primary").hide();
+                        }
+                         $('#id_site_code').attr('readonly', selectedSiteCode !== original_site_code);
+                         $('#id_site_name').attr('readonly', selectedSiteCode !== original_site_code);
+                         $('#id_elevation_m').attr('readonly', selectedSiteCode !== original_site_code);
+                         $('#id_elevation_datum').attr('readonly', selectedSiteCode !== original_site_code);
+                         $('#id_site_type').attr('readonly', selectedSiteCode !== original_site_code);
+                         $('#id_latitude').attr('readonly', selectedSiteCode !== original_site_code);
+                         $('#id_longitude').attr('readonly', selectedSiteCode !== original_site_code);
+                        if (selectedSiteCode !== original_site_code){
+                             $("#id_site_type").css('pointer-events', 'none');
+                             $("#id_elevation_datum").css('pointer-events', 'none');
+                         }
+                         else {
+                             $("#id_site_type").css('pointer-events', 'auto');
+                             $("#id_elevation_datum").css('pointer-events', 'auto');
+                        }
+                        return false;
+                    }
+                });
+            })
+        }
+        function processVariableMetadataElement(){
+            var variable_text = $('#id_available_variables').val();
+            var variables_json = $.parseJSON(variable_text);
+            var original_variable_code = $('#id_variable_code').val();
+            // remove the select option "Other.." from variable code selection dropdown
+            $("#id_variable_code_choices option[value='Other']").remove();
+            $('#id_variable_code_choices').change(function () {
+                var selectedVariableCode = this.value;
+                if(selectedVariableCode === "----"){
+                    $('#id-variable').find("button.btn-primary").hide();
+                    $('#id_variable_code').val("");
+                    $('#id_variable_code').attr('readonly', false);
+                    $('#id_variable_name').val("");
+                    $('#id_variable_name').attr('readonly', false);
+                    $('#id_variable_type').val("");
+                    $('#id_variable_type').attr('readonly', false);
+                    $('#id_no_data_value').val("");
+                    $('#id_no_data_value').attr('readonly', false);
+                    $('#id_variable_definition').val("");
+                    $('#id_variable_definition').attr('readonly', false);
+                    $('#id_speciation').val("");
+                    $('#id_speciation').attr('readonly', false);
```

Comment:
Same comment as above, to use array to bulk update attribute rather than duplicating the code.